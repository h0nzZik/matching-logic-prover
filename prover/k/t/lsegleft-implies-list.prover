claim \implies( \and( listSegmentLeft( variable("H") { ArrayIntInt } , variable("X") { Int } , variable("Y") { Int } , variable("F") { Set } , .Patterns) 
                    , list( variable("H") { ArrayIntInt }, variable("Y") { Int }, variable("G") { Int }, .Patterns) 
                    , \equals( variable("K") { Set }, union(variable("F") { Set }, variable("G") { Int })) 
                    , disjoint(variable("F") { Set }, variable("G") { Int }) 
                    , .Patterns )
              , \and( list( variable("H") { ArrayIntInt }, variable("X") { Int }, variable("K") { Set }, .Patterns), .Patterns )
              )
strategy search-bound(3)

/*
strategy kt(listSegmentLeft) ; ( ( simplify ; direct-proof )
                          | (right-unfold-Nth(0, 1) ; simplify ; direct-proof)
                        )
*/
