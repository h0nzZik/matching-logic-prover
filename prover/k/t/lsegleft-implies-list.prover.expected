<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( listSegmentLeft ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "F" ) { Set } , .Patterns ) , list ( variable ( "H" ) { ArrayIntInt } , variable ( "Y" ) { Int } , variable ( "G" ) { Int } , .Patterns ) , \equals ( variable ( "K" ) { Set } , union ( variable ( "F" ) { Set } , variable ( "G" ) { Int } ) ) , disjoint ( variable ( "F" ) { Set } , variable ( "G" ) { Int } ) , .Patterns ) , \and ( list ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "K" ) { Set } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
